### PR TITLE
Add Pipfiles with dependencies

### PR DESCRIPTION
**Pull request by AlinderS**
Pipfiles are created as standard by pipenv and contain the dependencies for running the project.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+ruff = "*"
```
Comment (eerovaher) [inline Pipfile:7]: Is Ruff a runtime dependency?

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+pycodestyle = "*"
```
Comment (eerovaher) [inline Pipfile:8]: We shouldn't be using `pycodestyle` anymore.

File: Pipfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+numpy = "*"
+matplotlib = "*"
+astropy = "*"
+rise = "*"
+
+[dev-packages]
+
+[requires]
+python_version = "3.11"
```
Comment (eerovaher) [inline Pipfile:17]: Why is only a single version of Python supported?

Comment (AlinderS) [reply Pipfile:17]: Pipenv automatically adds the currently used version as the only supported version, and there is no way to make it a range of versions. The simplest solution is to not require any specific version at all, so I will do that.

File: Pipfile (hunk 2 above)
Comment (AlinderS) [reply Pipfile:8]: Right, removing.

File: Pipfile (hunk 1 above)
Comment (AlinderS) [reply Pipfile:7]: It is being demonstrated in session 4, so I would say yes.

**Comment by AlinderS** [comment]
There are two missing dependencies if this is to be used, `black` should be a development dependency, and `tqdm` is a regular one.

**Review by eerovaher — COMMENTED** [review]
I don't use `pipenv` myself, so I can't do a proper review, but `Pipfile.lock` with its 2000 lines seems a bit excessive to me.

**Comment by AlinderS** [comment]
I am basing my decision to include the lockfile in the tree on [this thread](https://github.com/pypa/pipenv/issues/598). I personally don't care super much either way. I want to have the Pipfile in the tree because otherwise pipenv will generate one every time I enter the virtual environment, and it contains human-accessible information that is useful for other people. The lockfile will make sure one gets exactly the right versions (a deterministic build) which isn't that important to us.

**Comment by eerovaher** [comment]
This repository is meant to be a very basic introduction to Python. If there's anything here that depends critically on the exact version of dependencies then we are in deep trouble even if there is a `Pipfile.lock` that provides exactly those versions.

**Comment by AlinderS** [comment]
I have removed Pipfile.lock from the tree. I think with this change the PR is ready to merge, and I think it is best to squash and merge.

**Pull request merged**